chaos_monkey/network/network_scanner.py

```diff
@@ -27,48 +52,72 @@ def initialize(self):
         LOG.info("Found local IP addresses of the machine: %r", self._ip_addresses)
         # for fixed range, only scan once.
         if WormConfiguration.range_class is FixedRange:
-            self._ranges = [WormConfiguration.range_class(None)]
+            self._ranges = [WormConfiguration.range_class(fixed_addresses=WormConfiguration.range_fixed)]
         else:
             self._ranges = [WormConfiguration.range_class(ip_address)
                             for ip_address in self._ip_addresses]
         if WormConfiguration.local_network_scan:
             self._ranges += [FixedRange([ip_address for ip_address in get_ips_from_interfaces()])]
         LOG.info("Base local networks to scan are: %r", self._ranges)
 
-    def get_victim_machines(self, scan_type, max_find=5, stop_callback=None):
-        assert issubclass(scan_type, HostScanner)
+    def get_victim_machines(self, scan_type, max_find=5, scan_size=ITERATION_BLOCK_SIZE, stop_callback=None):
+        """
+        Generator of victim machines based on the network scanning configuration.
+        Every time scans
+        :param scan_type: scanner_class type
+        :param max_find:  Maximum number of victims to find
+        :param scan_size: Number of hosts to scan in parralel
+        :param stop_callback: Callback per iteration to check if we should stop activity
+        :return: Generates VictimHost instances
+        """
 
         scanner = scan_type()
         victims_count = 0
+        pool = ThreadPool(ITERATION_BLOCK_SIZE)
 
         for net_range in self._ranges:
             LOG.debug("Scanning for potential victims in the network %r", net_range)
-            for victim in net_range:
+            for scan_chunk in _grouper(net_range, scan_size):
                 if stop_callback and stop_callback():
                     LOG.debug("Got stop signal")
                     break
```

Comment:
return, not break. Now it will break out of the internal for-loop, though we want it to break out of the bigger one

Reply:
Fixed